presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java

```diff
@@ -146,6 +146,9 @@ public void testJsonToMap()
     public void testSubscript()
             throws Exception
     {
+        assertInvalidFunction("MAP(ARRAY [null], ARRAY [1])", "map key cannot be null");
```

Comment:
This is the wrong message to be checking for. The function shouldn't even resolve, since array[null] is of type array<unknown> and is therefore not comparable. Looks like there's a bug in the signature code that I'm fixing, where this is matching